File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {
    
    const set = new Set();
    const array = [];

    for(let i = 0; i < A.length; i++) {
        
        set.add(B[i]);
        let count = 0;

        for(let j = 0; j <= i; j++) {
            if(set.has(A[j])) count++;
        }
        array.push(count);
    }

    return array;
};

File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size();
        vector<int> freq(n + 1, 0);
        vector<int> preFixCommonArray;
        int count = 0;

        for(int idx = 0; idx < n; idx++) {
            int i = A[idx];
            int j = B[idx];

            freq[i]++;
            if(freq[i] == 2) count++;
            freq[j]++;
            if(freq[j] == 2) count++;
            preFixCommonArray.push_back(count);
        }
        return preFixCommonArray;
    }
};